lib/type_check.ex

```diff
@@ -322,4 +323,99 @@ defmodule TypeCheck do
       {:error, exception} -> raise exception
     end
   end
+  
+  @doc """
+  Ensure at runtime that arguments at result of function call conform the function spec.
+  
+  The function spec is extracted at compile time from the regular Elixir (oe Erlang) `@spec`.
+
+  ## Experimental
+  
+  This macro is experimental. Use it at your own risk only in a test-covered code.
+  If it explodes, please, [open an issue](https://github.com/Qqwy/elixir-type_check/issues).
+  
+  ## Examples
+  
+      iex> TypeCheck.ensure_types!(Kernel.abs(-13))
+      13
+      iex> TypeCheck.ensure_types!(Kernel.abs("hi"))
+      ** (TypeCheck.TypeError) At lib/type_check.ex:280:
+          `"hi"` is not a number.
+  """
+  @spec ensure_types!(Macro.t) :: Macro.t
```

Comment:
We ensure that the values have the correct types, hence the name is "ensure types". Or you can think about it as type-specific "ensure" from DbC. I'm open to better ideas but "well-typed" seems also confusing. What is even "well-typed"? Sounds like the annotations satisfy some expected type coverage or whatever.

Reply:
:thinking: You are right. 'Well-Typed' did not mean what I remembered it to mean. It actually means "a semantically correct type" (the inverse of 'ill-typed') rather than "a value conforming to its type".

I still think we should give it another name as `ensure_types!` makes it sound like we try to ensure that the input _is_ a type rather than _an expression adhering_ to its type.
When I think of a better name, I'll let you know.

Reply:
Maybe `enforce_types!`?